### ISSUE TITLE
Bump helm chart version alongside appVersion

### ISSUE THREAD

**Issue by mpalczynski** [Intility/helm-version#1]
Current implementation breaks the helm chart rule regarding chart versioning.
Comment from the Chart.yaml for `version` field:

```
# This is the chart version. This version number should be incremented each time you make changes
# to the chart and its templates, including the app version.
# Versions are expected to follow Semantic Versioning (https://semver.org/)
```

Therefore changing the app version should change the chart version as well.
I suggest adding a parameter named `sync` (true by default) to set the chart version the same as the appVersion.

**Comment by h3rmanj** [comment]
Hi! Thanks for pointing this out 😄 

While having a synced version would work in most cases, I think there might be some where `version` and `appVersion` is desynced on purpose. E.g. someone has explicitly set their `version` to `1.4.0`, and the `appVersion` is `1.3.0`. If they then bump via `npm version` with synced fields, the `version` field would be unchanged.

We could use [semver](https://www.npmjs.com/package/semver) to calculate the diff between the old and new `appVersion`, and then apply that to the `version` field to allow for desynced fields.

```js
import semverDiff from 'semver/functions/diff'
import semverInc from 'semver/functions/inc'

let diff = semverDiff(oldAppVersion, newAppVersion);
let newVersion = semverInc(diff);
```

Thoughts?

**Comment by mpalczynski** [comment]
Your solution seems to be more flexible and appropriate :) In my case, I had to keep the version equal. However, going with diff calculations we can achieve the same behavior and still be able to use maintain desynced versioning.

**Comment by h3rmanj** [comment]
Alright, would you like to have a go at it? 😄

**Comment by h3rmanj** [comment]
Hi @mpalczynski , sorry for leaving this open for so long. I've now released 1.3.0 which will bump the version field based on the type of version bump on the app version, as discussed.

If you haven't already found another solution, I'd love to hear if this works for you ☺️
